### PR TITLE
Add gas metering injection for grow_memory

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: src/gas.rs [hunk 1]
```diff
@@ -20,7 +20,50 @@ enum InjectAction {
 	IncrementSpawn,
 }
 
-pub fn inject_counter(opcodes: &mut elements::Opcodes, rules: &rules::Set, gas_func: u32) {
+fn inject_grow_counter(opcodes: &mut elements::Opcodes, total_funcs: u32) -> usize {
```
Comment (pepyakin) [inline src/gas.rs]: maybe it would be better to call `total_funcs` say `grow_counter_func`?

File: src/gas.rs [hunk 2]
```diff
@@ -91,7 +134,7 @@ pub fn inject_gas_counter(module: elements::Module, rules: &rules::Set) -> eleme
 			.build_sig()
 		);
 
-	let mut gas_func = mbuilder.push_import(
+	 mbuilder.push_import(
```
Comment (pepyakin) [inline src/gas.rs]: It looks there is stray space

**Comment by NikVolf** [comment]
@pepyakin fixed!

**Comment by pepyakin** [comment]
good to merge as soon as CI tests pass

**Pull request merged**